packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -1030,6 +1030,36 @@ class MemoryController extends DisposableController
     return serviceManager?.vm?.operatingSystem == 'android';
   }
 
+  /// All library dependencies for the app.
+  List<LibraryRef> get activeLibraries =>
+      serviceManager.isolateManager.selectedIsolateLibraries;
+
+  /// Source file name as returned from allocation's stacktrace.
+  /// Map source URI
+  ///    packages/flutter/src/widgets/image.dart
+  /// would map to
+  ///    package:flutter/lib/src/widgets/image.dart
+  // TODO(terry): Review with Ben pathing doesn't quite work the
+  //              source file has the lib/ maybe a LibraryRef could
+  //              be returned if in a library otherwise file:/// as today?
+  static const packageName = '/packages/';
+  String displayAsPackage(String sourceName) {
+    final packagesIndex = sourceName.indexOf(packageName);
+    final restToMatchStartIndex = packagesIndex + packageName.length;
+    if (packagesIndex >= 0) {
+      final match = activeLibraries.firstWhere(
```

Comment:
this seems slow and error prone. Can we just continue to display the full file path for now? Longer term we need to get an API in DDS that maps file paths to dart uris and vice versa.
Add a reference to 
https://github.com/dart-lang/sdk/issues/45530
to fix this code once there is a DDS api for this.

Reply:
Done.